## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 553966`
Commit: `f6d1eec977217d9f4f44cda24b1490c949772584`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>cwltool</li>
    <li>python313Packages.apiflask</li>
    <li>python313Packages.pyipv8</li>
    <li>python314Packages.apiflask</li>
    <li>python314Packages.pyipv8</li>
    <li>tribler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>graphinder</li>
    <li>kube-hunter</li>
    <li>python313Packages.aiohttp-apispec</li>
    <li>python313Packages.dingz</li>
    <li>python313Packages.ini2toml</li>
    <li>python313Packages.liccheck</li>
    <li>python313Packages.nats-python</li>
    <li>python313Packages.pkg-resources-backport</li>
    <li>python313Packages.python-velbus</li>
    <li>python313Packages.validate-pyproject</li>
    <li>python313Packages.validate-pyproject-schema-store</li>
    <li>python313Packages.webargs</li>
    <li>python314Packages.aiohttp-apispec</li>
    <li>python314Packages.dingz</li>
    <li>python314Packages.ini2toml</li>
    <li>python314Packages.liccheck</li>
    <li>python314Packages.nats-python</li>
    <li>python314Packages.pkg-resources-backport</li>
    <li>python314Packages.python-velbus</li>
    <li>python314Packages.validate-pyproject</li>
    <li>python314Packages.validate-pyproject-schema-store</li>
    <li>python314Packages.webargs</li>
  </ul>
</details>


---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>cwltool</summary>
<pre>SKIPPED [1] tests/test_ext.py:14: Requires the docker executable on the system path.
SKIPPED [1] tests/test_subgraph.py:325: Requires the docker executable on the system path.
SKIPPED [1] tests/test_dependencies.py:25: galaxy-tool-util is not installed
SKIPPED [1] tests/test_dependencies.py:47: galaxy-tool-util is not installed
SKIPPED [1] tests/test_dependencies.py:111: galaxy-tool-util is not installed
SKIPPED [1] tests/test_singularity.py:103: Requires the singularity executable on the system path.
SKIPPED [1] tests/test_singularity.py:118: Requires the singularity executable on the system path.
SKIPPED [1] tests/test_singularity.py:134: Requires that version 2.6.x of singularity executable version is on the system path.
SKIPPED [1] tests/test_singularity.py:158: Requires that version 3.x of singularity executable version is on the system path.
SKIPPED [1] tests/test_secrets.py:61: Requires the docker executable on the system path.
SKIPPED [1] tests/test_secrets.py:82: Requires the singularity executable on the system path.
SKIPPED [1] tests/test_iwdr.py:24: Requires the docker executable on the system path.
SKIPPED [1] tests/test_tmpdir.py:126: Requires the docker executable on the system path.
SKIPPED [1] tests/test_subgraph.py:289: Requires the docker executable on the system path.
SKIPPED [1] tests/test_subgraph.py:221: Requires the docker executable on the system path.
SKIPPED [1] tests/test_relocate.py:12: Requires the docker executable on the system path.
FAILED tests/test_mpi.py::test_singularity - schema_salad.exceptions.ValidationException: checking field &#x27;requirements&#x27;
FAILED tests/test_mpi.py::test_udocker - schema_salad.exceptions.ValidationException: checking field &#x27;requirements&#x27;
FAILED tests/test_mpi.py::test_docker_hint - schema_salad.exceptions.ValidationException: checking field &#x27;requirements&#x27;
=========== 3 failed, 270 passed, 144 skipped, 35 warnings in 16.73s ===========</pre>
</details>
<details>
<summary>python313Packages.apiflask</summary>
<pre>                &#x27;X-boolean&#x27;: Boolean(metadata={&#x27;description&#x27;: &#x27;A boolean header&#x27;}),
                &#x27;X-integer&#x27;: Integer(metadata={&#x27;description&#x27;: &#x27;An integer header&#x27;}),
&gt;               &#x27;X-number&#x27;: Number(metadata={&#x27;description&#x27;: &#x27;A number header&#x27;}),
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                &#x27;X-string&#x27;: String(metadata={&#x27;description&#x27;: &#x27;A string header&#x27;}),
                &#x27;X-array&#x27;: List(String(), metadata={&#x27;description&#x27;: &#x27;An array header&#x27;}),
            },
        )
E       TypeError: Can&#x27;t instantiate abstract class Number without an implementation for abstract method &#x27;num_type&#x27;

tests/test_openapi_headers.py:20: TypeError
=============================== warnings summary ===============================
tests/test_openapi_security.py::test_deprecated_apikey_auth_security_scheme
  /build/source/tests/test_openapi_security.py:49: DeprecationWarning: The API key authorization by HTTPTokenAuth is deprecated and will be removed in APIFlask 4.0.0. Use APIKeyHeaderAuth, APIKeyCookieAuth, or APIKeyQueryAuth instead.
    auth = HTTPTokenAuth(&#x27;apiKey&#x27;, header=&#x27;X-API-Key&#x27;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_openapi_headers.py::test_spec_with_dict_headers - TypeError: Can&#x27;t instantiate abstract class Number without an implementatio...
============= 1 failed, 513 passed, 1 skipped, 1 warning in 2.99s ==============</pre>
</details>
<details>
<summary>python313Packages.pyipv8</summary>
<pre>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ipv8/keyvault/crypto.py:111: in generate_key
    return M2CryptoSK(curve[0])
           ^^^^^^^^^^^^^^^^^^^^
ipv8/keyvault/private/m2crypto.py:35: in __init__
    super().__init__(ec_pub=ec.generate_private_key(curve, default_backend()))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
=============================== warnings summary ===============================
ipv8/messaging/anonymization/payload.py:340
  /build/source/ipv8/messaging/anonymization/payload.py:340: PytestCollectionWarning: cannot collect test class &#x27;TestRequestPayload&#x27; because it has a __init__ constructor (from: ipv8/test/messaging/anonymization/test_hiddenservices.py)
    @vp_compile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ipv8/test/keyvault/test_crypto.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
ERROR ipv8/test/peerdiscovery/test_network.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
ERROR ipv8/test/test_peer.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
========================= 1 warning, 3 errors in 0.56s =========================</pre>
</details>
<details>
<summary>python314Packages.apiflask</summary>
<pre>                &#x27;X-boolean&#x27;: Boolean(metadata={&#x27;description&#x27;: &#x27;A boolean header&#x27;}),
                &#x27;X-integer&#x27;: Integer(metadata={&#x27;description&#x27;: &#x27;An integer header&#x27;}),
&gt;               &#x27;X-number&#x27;: Number(metadata={&#x27;description&#x27;: &#x27;A number header&#x27;}),
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                &#x27;X-string&#x27;: String(metadata={&#x27;description&#x27;: &#x27;A string header&#x27;}),
                &#x27;X-array&#x27;: List(String(), metadata={&#x27;description&#x27;: &#x27;An array header&#x27;}),
            },
        )
E       TypeError: Can&#x27;t instantiate abstract class Number without an implementation for abstract method &#x27;num_type&#x27;

tests/test_openapi_headers.py:20: TypeError
=============================== warnings summary ===============================
tests/test_openapi_security.py::test_deprecated_apikey_auth_security_scheme
  /build/source/tests/test_openapi_security.py:49: DeprecationWarning: The API key authorization by HTTPTokenAuth is deprecated and will be removed in APIFlask 4.0.0. Use APIKeyHeaderAuth, APIKeyCookieAuth, or APIKeyQueryAuth instead.
    auth = HTTPTokenAuth(&#x27;apiKey&#x27;, header=&#x27;X-API-Key&#x27;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_openapi_headers.py::test_spec_with_dict_headers - TypeError: Can&#x27;t instantiate abstract class Number without an implementatio...
============= 1 failed, 513 passed, 1 skipped, 1 warning in 3.02s ==============</pre>
</details>
<details>
<summary>python314Packages.pyipv8</summary>
<pre>ipv8/keyvault/private/m2crypto.py:35: in __init__
    super().__init__(ec_pub=ec.generate_private_key(curve, default_backend()))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
=============================== warnings summary ===============================
ipv8/taskmanager.py:48: 10 warnings
  /build/source/ipv8/taskmanager.py:48: DeprecationWarning: &#x27;asyncio.iscoroutinefunction&#x27; is deprecated and slated for removal in Python 3.16; use inspect.iscoroutinefunction() instead
    if not iscoroutinefunction(func):

ipv8/messaging/anonymization/payload.py:340
  /build/source/ipv8/messaging/anonymization/payload.py:340: PytestCollectionWarning: cannot collect test class &#x27;TestRequestPayload&#x27; because it has a __init__ constructor (from: ipv8/test/messaging/anonymization/test_hiddenservices.py)
    @vp_compile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ipv8/test/keyvault/test_crypto.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
ERROR ipv8/test/peerdiscovery/test_network.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
ERROR ipv8/test/test_peer.py - cryptography.exceptions.UnsupportedAlgorithm: Curve sect163k1 is not supported
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 11 warnings, 3 errors in 0.56s ========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/dingz/default.nix
+++ b/pkgs/development/python-modules/dingz/default.nix
@@ -5,24 +5,31 @@
   buildPythonPackage,
   click,
   fetchFromGitHub,
+  pkg-resources-backport,
+  setuptools,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "dingz";
   version = "0.5.0";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "home-assistant-ecosystem";
     repo = "python-dingz";
-    rev = version;
+    tag = finalAttrs.version;
     hash = "sha256-bCytQwLWw8D1UkKb/3LQ301eDCkVR4alD6NHjTs6I+4=";
   };
 
-  propagatedBuildInputs = [
+  pythonRelaxDeps = [ "async_timeout" ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
     aiohttp
     async-timeout
     click
+    pkg-resources-backport
   ];
 
   # Project has no tests
@@ -32,9 +39,10 @@ buildPythonPackage rec {
 
   meta = {
     description = "Python API for interacting with Dingz devices";
-    mainProgram = "dingz";
     homepage = "https://github.com/home-assistant-ecosystem/python-dingz";
+    changelog = "https://github.com/home-assistant-ecosystem/python-dingz/releases/tag/${finalAttrs.src.tag}";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ fab ];
+    mainProgram = "dingz";
   };
-}
+})

--- a/pkgs/development/python-modules/validate-pyproject-schema-store/default.nix
+++ b/pkgs/development/python-modules/validate-pyproject-schema-store/default.nix
@@ -1,0 +1,43 @@
+{
+  lib,
+  buildPythonPackage,
+  fastjsonschema,
+  fetchFromGitHub,
+  hatchling,
+  nix-update-script,
+  pytestCheckHook,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "validate-pyproject-schema-store";
+  version = "2026.08.15";
+  pyproject = true;
+
+  __structuredAttrs = true;
+
+  src = fetchFromGitHub {
+    owner = "henryiii";
+    repo = "validate-pyproject-schema-store";
+    tag = finalAttrs.version;
+    hash = "sha256-iiBTEMP+vhvOQFS500LhBuf2epEahqN+jHheHUR5kCA=";
+  };
+
+  build-system = [ hatchling ];
+
+  dependencies = [ fastjsonschema ];
+
+  # Circular dependency with validate-pyproject
+  doCheck = false;
+
+  pythonImportsCheck = [ "validate_pyproject_schema_store" ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Daily automatic mirror of SchemaStore for validate-pyproject";
+    homepage = "https://github.com/henryiii/validate-pyproject-schema-store";
+    changelog = "https://github.com/henryiii/validate-pyproject-schema-store/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ fab ];
+  };
+})

--- a/pkgs/development/python-modules/validate-pyproject/default.nix
+++ b/pkgs/development/python-modules/validate-pyproject/default.nix
@@ -1,0 +1,96 @@
+{
+  lib,
+  buildPythonPackage,
+  fastjsonschema,
+  fetchFromGitHub,
+  nix-update-script,
+  packaging,
+  pytest-cov-stub,
+  pytestCheckHook,
+  setuptools-scm,
+  setuptools,
+  tomli,
+  trove-classifiers,
+  validate-pyproject-schema-store,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "validate-pyproject";
+  version = "0.25";
+  pyproject = true;
+
+  __structuredAttrs = true;
+
+  src = fetchFromGitHub {
+    owner = "abravalheri";
+    repo = "validate-pyproject";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-byxghU2x8at4tAAOzIzlnnsMbvzer//0R0DNiIE4Dpk=";
+  };
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [ fastjsonschema ];
+
+  optional-dependencies = {
+    all = [
+      packaging
+      tomli
+      trove-classifiers
+    ];
+    store = [ validate-pyproject-schema-store ];
+  };
+
+  nativeCheckInputs = [
+    pytest-cov-stub
+    pytestCheckHook
+  ]
+  ++ lib.flatten (builtins.attrValues finalAttrs.passthru.optional-dependencies);
+
+  pythonImportsCheck = [ "validate_pyproject" ];
+
+  disabledTests =
+    let
+      examples = [
+        "cibuildwheel/default.toml"
+        "cibuildwheel/overrides.toml"
+        "pdm/pyproject.toml"
+        "poetry/poetry-inline-table.toml"
+        "poetry/poetry-author-no-email.toml"
+        "poetry/poetry-readme-files.toml"
+        "poetry/poetry-sample-project.toml"
+        "poetry/poetry-complete.toml"
+        "poetry/poetry-capital-in-author-email.toml"
+        "ruff/modern.toml"
+        "store/example.toml"
+      ];
+    in
+    [
+      # Tests require network access
+      "test_cache_open_url"
+      "test_invalid_examples_api"
+      "test_invalid_examples_cli"
+      "TestClassifiers::test_downloaded"
+      "test_list_from_entry_points"
+    ]
+    ++ lib.concatMap (example: [
+      "test_examples_api[${example}]"
+      "test_examples_cli[${example}]"
+      "test_examples_api[${example}-api_pre_compile]"
+      "test_examples_api[${example}-cli_pre_compile]"
+    ]) examples
+    ++ [ "test_downloaded" ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Validation library for simple check on pyproject.toml";
+    homepage = "https://github.com/abravalheri/validate-pyproject";
+    changelog = "https://github.com/abravalheri/validate-pyproject/blob/${finalAttrs.src.rev}/CHANGELOG.rst";
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ fab ];
+  };
+})

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -22058,6 +22058,12 @@ self: super: with self; {
 
   validate-email = callPackage ../development/python-modules/validate-email { };
 
+  validate-pyproject = callPackage ../development/python-modules/validate-pyproject { };
+
+  validate-pyproject-schema-store =
+    callPackage ../development/python-modules/validate-pyproject-schema-store
+      { };
+
   validator-collection = callPackage ../development/python-modules/validator-collection { };
 
   validators = callPackage ../development/python-modules/validators { };

